### PR TITLE
Better weather icons

### LINKED ISSUE #31: Rain Icon location seems wrong
I don't know if this is an actual bug but it seems like the rainy day sprite that's being pulled isn't the right part.
It just seems like a blob instead of a cloud/dark cloud.

image

### PR DESCRIPTION
**Pull request by drewhoener**
Instead of having weird cropped parts from the Animations sheet, changed the upcoming weather icons to the ones on the tv (Cursors sheet). 

I also went a bit overboard, the TV has this nice border that I wanted around the icons so I added a few methods and some logic to create a custom sheet with those parts stitched together

The TV border 
image

The new icons
image

Fixes #31

### CODE REVIEW

**Comment by Annosz** [comment]
Ah, thank you very much, I just couldn't do it! I also tried getting the weather icons from the TV, but they were too small, and I did not know how to create the custom sheet, so I went with the weird clouds, but I didn't like it either...

**Comment by Zamiell** [comment]
very nice work drew

File: UIInfoSuite2/Infrastucture/Tools.cs [hunk 1]
```diff
@@ -114,5 +114,41 @@ public static Item GetHoveredItem()
 
             return hoverItem;
         }
+
+        public static void GetSubTexture(Color[] output, Color[] originalColors, Rectangle sourceBounds, Rectangle clipArea)
+        {
+            //Create output array
```
Comment (Annosz) [inline UIInfoSuite2/Infrastucture/Tools.cs:120]: I think this comment is quite misleading and should be deleted.

File: UIInfoSuite2/UIElements/ShowRainyDayIcon.cs [hunk 2]
```diff
@@ -82,8 +87,54 @@ private void OnRenderedHud(object sender, RenderedHudEventArgs e)
...
+            ModEntry.MonitorObject.Log("Setting up icon sheet", LogLevel.Info);
+            // Setup Texture sheet as a copy, so as not to disturb existing sprites
+            _iconSheet = new Texture2D(Game1.graphics.GraphicsDevice , WeatherSheetWidth, WeatherSheetHeight);
+            _weatherIconColors = new Color[WeatherSheetWidth * WeatherSheetHeight];
+            var cursorColors = new Color[Game1.mouseCursors.Width * Game1.mouseCursors.Height];
+            var bounds = new Rectangle(0, 0, Game1.mouseCursors.Width, Game1.mouseCursors.Height);
+            Game1.mouseCursors.GetData(cursorColors);
+            var subTextureColors = new Color[15 * 15];
+            
+            //ModEntry.MonitorObject.Log($"Got tile sheet with length {cursorColors.Length}", LogLevel.Alert);
```
Comment (Annosz) [inline UIInfoSuite2/UIElements/ShowRainyDayIcon.cs:108]: Leftover comment from testing.

File: UIInfoSuite2/UIElements/ShowRainyDayIcon.cs [hunk 3]
```diff
@@ -82,8 +87,54 @@ private void OnRenderedHud(object sender, RenderedHudEventArgs e)
...
+            // Copy over the bits we want
+            // Border from TV screen
+            Tools.GetSubTexture(subTextureColors, cursorColors, bounds, new Rectangle(499, 307, 15, 15));
+            // Copy to each destination
+            Tools.SetSubTexture(subTextureColors, _weatherIconColors, WeatherSheetWidth, new Rectangle(0, 0, 15, 15));
+            Tools.SetSubTexture(subTextureColors, _weatherIconColors, WeatherSheetWidth, new Rectangle(15, 0, 15, 15));
+            Tools.SetSubTexture(subTextureColors, _weatherIconColors, WeatherSheetWidth, new Rectangle(30, 0, 15, 15));
+            //ModEntry.MonitorObject.Log("Got TV textures", LogLevel.Alert);
```
Comment (Annosz) [inline UIInfoSuite2/UIElements/ShowRainyDayIcon.cs:116]: Leftover comment from testing.

File: UIInfoSuite2/UIElements/ShowRainyDayIcon.cs [hunk 4]
```diff
@@ -82,8 +87,54 @@ private void OnRenderedHud(object sender, RenderedHudEventArgs e)
...
+            
+            // Stormy Weather
+            Tools.GetSubTexture(subTextureColors, cursorColors, bounds, new Rectangle(426, 346, 13, 13));
+            Tools.SetSubTexture(subTextureColors, _weatherIconColors, WeatherSheetWidth,  new Rectangle(16, 1, 13, 13));
+            
+            // Snowy Weather
+            Tools.GetSubTexture(subTextureColors, cursorColors, bounds, new Rectangle(465, 346, 13, 13));
+            Tools.SetSubTexture(subTextureColors, _weatherIconColors, WeatherSheetWidth,  new Rectangle(31, 1, 13, 13));
+            
+            //ModEntry.MonitorObject.Log("Got weather textures, setting icon sheet...", LogLevel.Alert);
```
Comment (Annosz) [inline UIInfoSuite2/UIElements/ShowRainyDayIcon.cs:131]: Leftover comment from testing.

File: UIInfoSuite2/UIElements/ShowRainyDayIcon.cs [hunk 5]
```diff
@@ -95,19 +146,21 @@ private void GetWeatherIconSpriteLocation()
 
                 case Game1.weather_rain:
                     _IsNextDayRainy = true;
-                    _weatherIconSpriteLocation = new Rectangle(268, 1750, 20, 20);
+                    // _weatherIconSpriteLocation = new Rectangle(268, 1750, 20, 20);
```
Comment (Annosz) [inline UIInfoSuite2/UIElements/ShowRainyDayIcon.cs:149]: You can delete this, no need to keep old codes.

File: UIInfoSuite2/UIElements/ShowRainyDayIcon.cs [hunk 6]
```diff
@@ -95,19 +146,21 @@ private void GetWeatherIconSpriteLocation()
...
+                    _weatherIconSpriteLocation = new Rectangle(0, 0, 15, 15);
                     _hoverText = _helper.SafeGetString(LanguageKeys.RainNextDay);
                     break;
 
                 case Game1.weather_lightning:
                     _IsNextDayRainy = true;
-                    _weatherIconSpriteLocation = new Rectangle(272, 1641, 20, 20);
+                    // _weatherIconSpriteLocation = new Rectangle(272, 1641, 20, 20);
```
Comment (Annosz) [inline UIInfoSuite2/UIElements/ShowRainyDayIcon.cs:156]: You can delete this, no need to keep old codes.

**Review by Annosz — CHANGES_REQUESTED** [review]
I just added a few request to delete the leftover comments, otherwise it's very nice 👍

File: UIInfoSuite2/Infrastucture/Tools.cs (hunk 1 above)
Comment (drewhoener) [reply UIInfoSuite2/Infrastucture/Tools.cs:120]: yeah it slipped through from when I was writing the method in a different way, I'll get rid of all the comments

**Review by Annosz — APPROVED** [review]

**Pull request merged**